src/libs/Network.js

```diff
@@ -95,15 +98,33 @@ Onyx.connect({
     callback: val => email = val ? val.email : null,
 });
 
+/**
+ *
+ * @param {Boolean} val
+ */
+function setIsReady(val) {
+    isReady = val;
+}
+
 /**
  * Checks to see if a request can be made.
  *
  * @param {Object} request
+ * @param {String} request.type
+ * @param {String} request.command
  * @param {Object} request.data
  * @param {Boolean} request.data.forceNetworkRequest
  * @return {Boolean}
  */
 function canMakeRequest(request) {
+    // We'll not make the request now if the network is not ready (it should have higher priority than
+    // forceNetworkRequest and isQueuePaused because we wait until credentials + session objects have
+    // been read from persistent storage).
+    if (!isReady) {
+        LogUtil.hmmm('Trying to make a request when Network is not ready', {command: request.command, type: request.type});
```

Comment:
Hehe yeah, I think it can help us to know which command is being called before auth + credentials are read from Onyx. Also in an ideal scenario that log shouldn't be called in unit tests (I added a couple of `expect` statements to check that message is not logged if auth token + credentials are already read).

Reply:
So, what I was getting at here is that...

1. We should use a constant whenever referencing something like an error message in multiple places. That way if the content changes later the reference won't.
2. Logs are not great things to test against because people can decide to remove them and the tests that are expecting them "not to be called" are going to keep passing anyway.

Reply:
Ah right, for the point 2, I removed the tests that check if the Log was not called with that message.

Reply:
NAB, `request.type` can probably be removed it's almost always going to be a `post` and not really relevant